syntax = "proto3";

package tenant_parser;

import "google/protobuf/empty.proto";

option go_package = "github.com/milkyhoop/tenant_parser/internal/delivery/pb;intentparser";

service IntentParserService {
  // Legacy methods (keep for compatibility)
  rpc DoSomething (IntentParserRequest) returns (IntentParserResponse);
  rpc HealthCheck (google.protobuf.Empty) returns (google.protobuf.Empty);
  
  // Core confidence engine methods
  rpc CalculateConfidence (ConfidenceRequest) returns (ConfidenceResponse);
  rpc MakeDecision (DecisionRequest) returns (DecisionResponse);
  
  // Utility methods
  rpc ExtractFaqAnswer (FaqExtractionRequest) returns (FaqExtractionResponse);
  rpc GetPoliteDeflection (DeflectionRequest) returns (DeflectionResponse);
}

// Legacy messages
message IntentParserRequest {
  string user_id = 1;
  string input = 2;
}

message IntentParserResponse {
  string status = 1;
  string result = 2;
}

// Core confidence engine messages
message ConfidenceRequest {
  string query = 1;
  string tenant_id = 2;
  repeated FaqResult faq_results = 3;
}

message FaqResult {
  string question = 1;
  string answer = 2;
  string content = 3;
  double similarity_score = 4;
  double score = 5;
}

message ConfidenceResponse {
  double confidence = 1;
  string tier_name = 2;
  string route = 3;
  double cost_per_query = 4;
  string intelligence_level = 5;
  int32 tier_number = 6;
  int32 tier = 7;
  bool api_call_required = 8;
  int32 faq_count = 9;
  string model = 10;
}

message DecisionRequest {
  double confidence = 1;
}

message DecisionResponse {
  string route = 1;
  int32 tier = 2;
  bool api_call = 3;
  string model = 4;
  double cost_per_query = 5;
  int32 faq_count = 6;
  string intelligence_level = 7;
  int32 response_time_ms = 8;
  bool api_call_required = 9;
}

// Utility messages
message FaqExtractionRequest {
  string faq_content = 1;
}

message FaqExtractionResponse {
  string extracted_answer = 1;
  string answer = 2;               // ADDED: Server assigns line 214
  string question = 3;             // ADDED: Server assigns line 215
  double confidence = 4;           // ADDED: Server assigns line 216
  bool direct_response = 5;        // ADDED: Server assigns line 217
}

message DeflectionRequest {
  string tenant_id = 1;
  string context = 2;
}

message DeflectionResponse {
  string deflection_message = 1;
  string message = 2;
  bool is_deflection = 3;
  string suggested_contact = 4;    // ADDED: Server assigns line 244
}